backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionServiceImpl.java

```diff
@@ -337,26 +337,47 @@ public void getLatestDatasetVersionByDatasetId(
             Any.pack(GetLatestDatasetVersionByDatasetId.Response.getDefaultInstance()));
       }
 
-      // TODO: Implement sort key in future
-      /*String sortKey =
-      request.getSortKey().isEmpty() ? ModelDBConstants.TIME_LOGGED : request.getSortKey();*/
-
-      int pageNumber = request.getAscending() ? 2 : 1;
-      DatasetVersionDTO datasetVersionDTO =
-          getDatasetVersionDTOByDatasetId(
-              request.getDatasetId(), pageNumber, 1, request.getAscending());
+      UserInfo currentLoginUserInfo = authService.getCurrentLoginUserInfo();
 
-      if (datasetVersionDTO.getDatasetVersions().size() != 1) {
+      FindRepositoriesBlobs.Builder findRepositoriesBlobs =
+          FindRepositoriesBlobs.newBuilder()
+              .addRepoIds(Integer.parseInt(request.getDatasetId()))
+              .setPageLimit(1)
+              .setPageNumber(1);
+      String sortKey =
+          request.getSortKey().isEmpty() ? ModelDBConstants.DATE_CREATED : request.getSortKey();
+      CommitPaginationDTO commitPaginationDTO =
+          commitDAO.findCommits(
+              findRepositoriesBlobs.build(),
+              currentLoginUserInfo,
+              false,
+              false,
+              true,
+              sortKey,
+              request.getAscending());
+      if (commitPaginationDTO.getCommitEntities().size() != 1) {
         logAndThrowError(
             "No datasetVersion found for dataset '" + request.getDatasetId() + "'",
             Code.NOT_FOUND_VALUE,
             Any.pack(GetLatestDatasetVersionByDatasetId.Response.getDefaultInstance()));
       }
 
+      RepositoryEntity repositoryEntity = null;
+      if (!request.getDatasetId().isEmpty()) {
+        RepositoryIdentification repositoryIdentification =
+            RepositoryIdentification.newBuilder()
+                .setRepoId(Long.parseLong(request.getDatasetId()))
+                .build();
+        repositoryEntity =
+            repositoryDAO.getProtectedRepositoryById(repositoryIdentification, false);
+      }
+
+      DatasetVersion datasetVersion =
+          convertRepoDatasetVersions(repositoryEntity, commitPaginationDTO.getCommits()).get(0);
       /*Build response*/
       responseObserver.onNext(
           GetLatestDatasetVersionByDatasetId.Response.newBuilder()
-              .setDatasetVersion(datasetVersionDTO.getDatasetVersions().get(0))
+              .setDatasetVersion(datasetVersion)
```

Comment:
If `convertRepoDatasetVersions` return list is empty then `get(0)` is returns null and protobuf not accept null value so here it will throw an exception.